bfvmm/include/vmcs/vmcs_intel_x64.h

```diff
@@ -455,142 +455,109 @@ class vmcs_intel_x64
 };
 
 inline auto
-get_vmcs_field(uint64_t addr, const std::string &name, bool exists)
+get_vmcs_field(uint64_t addr, const char *name, bool exists)
 {
-    auto what = std::string("get_vmcs_field failed: ") + name + " field doesn't exist";
-
     if (!exists)
-        throw std::logic_error(what);
+        throw std::logic_error("get failed: field doesn't exist");
 
     return intel_x64::vm::read(addr, name);
 }
 
 inline auto
-get_vmcs_field_if_exists(uint64_t addr, const std::string &name, bool verbose, bool exists)
+get_vmcs_field_if_exists(uint64_t addr, const char *name, bool verbose, bool exists)
 {
-    if (!exists && verbose)
-        bfwarning << "get_vmcs_field_if_exists failed: " << name << " field doesn't exist" << '\n';
-
     if (exists)
         return intel_x64::vm::read(addr, name);
 
+    if (!exists && verbose)
+        bfwarning << "get failed: " << name << " doesn't exist" << '\n';
```

Comment:
same thing here. feel free to provide the function name. slow is fine on the error case